torch/jit/_monkeytype_config.py

```diff
@@ -0,0 +1,115 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+# mypy: ignore-errors
+
+from typing import Optional, Iterable, List, Dict
+from collections import defaultdict
+
+_IS_MONKEYTYPE_INSTALLED = True
+try:
+    import monkeytype
+    from monkeytype import trace as monkeytype_trace
+    from monkeytype.db.base import CallTraceStore, CallTraceStoreLogger
+    from monkeytype.config import default_code_filter
+    from monkeytype.tracing import CallTrace, CodeFilter
+except ImportError:
+    _IS_MONKEYTYPE_INSTALLED = False
+    print("Warning: monkeytype is not installed. Please install monkeytype to enable Profile-Directed Typing in TorchScript")
+
+def get_qualified_name(func):
+    return func.__qualname__
+
+if _IS_MONKEYTYPE_INSTALLED:
+
+    class JitTypeTraceStoreLogger(CallTraceStoreLogger):
+        """A JitTypeCallTraceLogger that stores logged traces in a CallTraceStore."""
+        def __init__(self, store: CallTraceStore):
+            super().__init__(store)
+
+        def log(self, trace: CallTrace) -> None:
+            self.traces.append(trace)
```

Comment:
Do we need this class?

Reply:
Yes, this class is required to log traces for those functions inside main.

Reply:
For functions defined in main, MonkeyType ignores the traces. Hence the traces will be lost for those function/method calls if we do not overload this class

Reply:
How does it work? `JitTypeTraceStoreLogger` writes traces to `self.traces`, which is not accessed during `analyze.

Reply:
It is not accessed during `analyze`. But `self.traces` are then used to populate `self.trace_records` which is used here https://github.com/pytorch/pytorch/pull/54513/files#diff-b16a3d89e6036d271793a297aea1cce21fa27f6dcc449206e9d1113dc3866688R38

Reply:
Ah I see! thanks for clarification